onnxruntime/core/optimizer/matmul_integer_to_float.cc

```diff
@@ -63,8 +81,11 @@ Status MatMulIntegerToFloatFusion::ApplyImpl(Graph& graph, bool& modified, int g
     auto& mul_node = *node_ptr;
 
     ORT_RETURN_IF_ERROR(Recurse(mul_node, modified, graph_level, logger));
-
-    if (!graph_utils::IsSupportedOptypeVersionAndDomain(mul_node, "Mul", {7, 13, 14})) {
+    auto ep_type = node_ptr->GetExecutionProviderType();
+    bool dml_ep = ep_type == kDmlExecutionProvider;
+    if (!graph_utils::IsSupportedOptypeVersionAndDomain(mul_node, "Mul", {7, 13, 14}) ||
+        !graph_utils::IsSupportedProvider(mul_node, GetCompatibleExecutionProviders()) ||
+        (!dml_ep && HasElementDataType(*mul_node.InputDefs()[0], ONNX_NAMESPACE::TensorProto_DataType_FLOAT16))) {
```

Comment:
This fusion won't work for DML EP as this is a L2 transformer, `graph_transformer_utils.cc` should register this fusion for DML EP as well.
https://github.com/microsoft/onnxruntime/blob/8d48d3e9cce4a7b5640051899f824c69a6a67c62/onnxruntime/core/optimizer/graph_transformer_utils.cc#L297C50-L297C60